rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -79,9 +79,13 @@ rclcpp::QoS publisher_qos_for_topic(
     return Rosbag2QoS{};
   }
 
-  const auto profiles_yaml = YAML::Load(topic.offered_qos_profiles);
-  const auto offered_qos_profiles = profiles_yaml.as<std::vector<Rosbag2QoS>>();
-  return Rosbag2QoS::adapt_offer_to_recorded_offers(topic.name, offered_qos_profiles);
+  std::vector<Rosbag2QoS> casted;
```

Comment:
Here, a conversion from `std::vector<rclcpp::QoS>` & `std::vector<rosbag2_storage::Rosbag2QoS>` is needed.